File: frontend/src/microservices/content/objects/BlogPostSummaryResponse.ts
import type { ContentImageResponse } from './ContentImageResponse'

export type BlogTagResponse = {
  tagType: string
  tagValue: string
}

export type BlogPostSummaryResponse = {
  postId: string
  authorUserId: string
  authorDisplayName: string
  authorAvatarUrl: string | null
  title: string
  summary: string
  coverImageUrl: string | null
  coverText: string
  travelCity: string | null
  travelCities: string[]
  status: string
  createdAt: string
  updatedAt: string
  publishedAt: string | null
  commentCount: number
  likeCount: number
  favoriteCount: number
  likedByCurrentUser: boolean
  favoritedByCurrentUser: boolean
  isMyPost: boolean
  canEdit: boolean
  canArchive: boolean
  images: ContentImageResponse[]
  tags: BlogTagResponse[]
  searchResultSnippet: string | null
}

export const blogPostSummaryResponseFromJson = (json: string): BlogPostSummaryResponse =>
  JSON.parse(json) as BlogPostSummaryResponse

export const blogPostSummaryResponseToJson = (value: BlogPostSummaryResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/blog/objects/BlogPostSummaryResponse.scala
// BlogPostSummaryResponse：博客域博客文章摘要返回对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BlogPostSummaryResponse(
    postId: String,
    authorUserId: String,
    authorDisplayName: String,
    authorAvatarUrl: Option[String],
    title: String,
    summary: String,
    coverImageUrl: Option[String],
    coverText: String,
    travelCity: Option[String],
    travelCities: List[String],
    status: String,
    createdAt: String,
    updatedAt: String,
    publishedAt: Option[String],
    commentCount: Long,
    likeCount: Long,
    favoriteCount: Long,
    likedByCurrentUser: Boolean,
    favoritedByCurrentUser: Boolean,
    isMyPost: Boolean,
    canEdit: Boolean,
    canArchive: Boolean,
    images: List[BlogImagePlannerResponse],
    tags: List[BlogTagPlannerResponse],
    searchResultSnippet: Option[String]
)
object BlogPostSummaryResponse:
  given sourceEncoder: Encoder[BlogPostSummaryResponse] = deriveEncoder
  given sourceDecoder: Decoder[BlogPostSummaryResponse] = deriveDecoder